src/core/jit/store.rs

```diff
@@ -1,22 +1,55 @@
+use std::collections::HashMap;
+
 use crate::core::jit::model::FieldId;
 
+/// Path to the data in the store with info
+/// to resolve nested multiple data
+#[derive(Debug, Clone)]
+pub struct DataPath {
+    /// List of paths, where every entry contains info about specific
+    /// level ot multiple data
+    multiple_path: Vec<usize>,
+}
+
+impl DataPath {
+    /// Create default DataPath that resolved to single value
+    pub fn single() -> Self {
+        Self { multiple_path: Vec::new() }
+    }
+
+    /// Create new DataPath with specified additional entry
+    pub fn with_index(&self, index: usize) -> Self {
+        let mut multiple_path = self.multiple_path.clone();
+
+        multiple_path.push(index);
+
+        Self { multiple_path }
+    }
+
+    /// Iterator over indexes only for multiple paths.
+    /// Helpful when collecting the data after it has been previously
+    /// resolved
+    pub fn multiple_indexes(&self) -> impl Iterator<Item = &usize> + '_ {
```

Comment:
```suggestion
    pub fn as_slice(&self) -> &[usize] {
```
Can we do this instead?

Reply:
yes, replaced